tests/api/db/test_functions.py

```diff
@@ -320,3 +300,61 @@ def test_delete_function(db: DBInterface, db_session: Session):
 
     assert number_of_tags == 0
     assert number_of_labels == 0
+
+
+@pytest.mark.parametrize(
+    "db,db_session", [(dbs[0], dbs[0])], indirect=["db", "db_session"]
+)
+@pytest.mark.parametrize("use_hash_key", [True, False])
+def test_list_functions_multiple_tags(
+    db: DBInterface, db_session: Session, use_hash_key: bool
+):
+    function_1 = {"bla": "blabla", "status": {"bla": "blabla"}}
+    function_name_1 = "function_name_1"
+
+    tags = ["some_tag", "some_tag2", "some_tag3"]
+    for tag in tags:
+        function_hash_key = db.store_function(
+            db_session, function_1, function_name_1, tag=tag, versioned=True
+        )
+    functions = db.list_functions(
+        db_session,
+        function_name_1,
+        hash_key=function_hash_key if use_hash_key else None,
+    )
+    assert len(functions) == len(tags)
+    for function in functions:
+        if use_hash_key:
+            assert function["metadata"]["hash"] == function_hash_key
+        function_tag = function["metadata"]["tag"]
+        tags.remove(function_tag)
+    assert len(tags) == 0
+
+
+@pytest.mark.parametrize(
+    "db,db_session", [(dbs[0], dbs[0])], indirect=["db", "db_session"]
+)
+def test_list_function_with_tag_and_uid(db: DBInterface, db_session: Session):
+    function_1 = {"bla": "blabla", "status": {"bla": "blabla"}}
+    function_2 = {"blabla": "bla", "status": {"blabla": "bla"}}
```

Comment:
lets create actual function objects as I have https://github.com/mlrun/mlrun/pull/3096 in the queue, which actually adds validation and masking when storing functions. Although my addition is on the crud layer one above this one it will be a good practice to get aligned on this

Reply:
same in the test above

Reply:
also, use meaningful content, when and if possible.